src/libratbag.c

```diff
@@ -511,6 +562,30 @@ ratbag_register_driver(struct ratbag *ratbag, struct ratbag_driver *driver)
 	list_insert(&ratbag->drivers, &driver->link);
 }
 
+static int
+open_restricted(const char *path, int flags, void *user_data)
+{
+	int fd = open(path, flags);
+
+	if (fd < 0)
+		fprintf(stderr, "Failed to open %s (%s)\n",
+			path, strerror(errno));
+
+	return fd < 0 ? -errno : fd;
+}
+
+static void
+close_restricted(int fd, void *user_data)
+{
+	close(fd);
+}
+
+LIBRATBAG_EXPORT const struct ratbag_interface
+ratbag_default_interface = {
+	.open_restricted = open_restricted,
+	.close_restricted = close_restricted,
+};
+
```

Comment:
I don't think we should export this. Either default to that if the interface is NULL or provide a pseudo-pointer in the form of `#define RATBAG_DEFAULT_INTERFACE (const struct ratbag_interface*)0x2`

Reply:
Yeah, the lazy me only took the quickest way by duplicating the ratbag-command.c code and put it in libratbag so that it gets exported by swig (and not add an other python dependency).

I exported it because we have a test case for it, and I think the idea of the `RATBAG_DEFAULT_INTERFACE` would solve this nicely.